src/foam/nanos/cron/CronScheduleDAO.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.cron',
+  name: 'CronScheduleDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: 'Update cron scheduledTime on schedule changed',
+
+  javaImports: [
+    'foam.nanos.cron.Cron',
+    'foam.util.SafetyUtil'
+  ],
+
+  methods: [
+    {
+      name: 'put_',
+      javaCode: `
+        Cron newCron = (Cron) obj;
+        Cron oldCron = (Cron) getDelegate().find_(x, obj);
+
+        if ( oldCron == null
+          || ! SafetyUtil.equals(oldCron.getSchedule(), newCron.getSchedule())
```

Comment:
Should this be `! SafetyUtil.equals(oldCron.getScheduledTime(), newCron.getScheduledTime()` instead?

Reply:
No, I want to re-calculate the `scheduledTime` only when the cron.schedule is changed.

Reply:
After a quick discussion this makes sense - the scheduledTime value was not being updated after a schedule is changed. This means that if the old schedule's next scheduled time was a date further in the future than the new schedule, CronScheduler.java would not update scheduledTime until the previously calculated time.